### PR TITLE
Add heuristic complexity scorer and golden trace tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- implement `HeuristicComplexity` combining depth and ambiguity heuristics
- add golden-file integration test suite with 50 queries
- document golden file design and mark roadmap task complete

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie`

------
https://chatgpt.com/codex/tasks/task_e_68c38f43a6588322a67d97f53cd425a5

## Summary by Sourcery

Introduce a baseline heuristic complexity scorer and a comprehensive golden-file test suite to validate its behavior, along with associated documentation updates and minor enhancements.

New Features:
- Add HeuristicComplexity provider that combines depth and ambiguity heuristics into a ComplexityFn implementation
- Include a golden-file integration test suite with ~50 curated queries to verify complexity traces against stored snapshots

Enhancements:
- Derive Debug and Clone on DepthHeuristic and AmbiguityHeuristic and export HeuristicComplexity from the main crate
- Adjust CLI trace output precision from four to three decimal places

Documentation:
- Convert inline citation markers to numbered footnotes in the complexity design document
- Mark the golden-file test suite task as complete in the roadmap and document its format and storage location

Tests:
- Add tests to validate golden-file loading, handling of missing/empty/malformed files, and integrity of trace invariants

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai pause

File: tests/golden_traces.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+fn golden_traces() {
+    let file = File::open("tests/golden/traces.jsonl").expect("missing golden file");
+    let reader = BufReader::new(file);
+    let hc = HeuristicComplexity::default();
+
+    for line in reader.lines() {
+        let line = line.expect("read line");
+        let expected: Trace = serde_json::from_str(&line).expect("parse trace");
+        let actual = hc.trace(&expected.query).expect("trace error");
+        assert_eq!(actual, expected, "mismatch for query `{}`", expected.query);
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/golden_traces.rs]: Has this now been resolved?

File: tests/golden_traces.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+use lag_complexity::{ComplexityFn, HeuristicComplexity, Trace};
+use rstest::rstest;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+
+#[rstest]
+#[expect(clippy::expect_used, reason = "test should fail loudly")]
+fn golden_traces() {
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/golden_traces.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@sourcery-ai review
